BoxContentSDK/BoxContentSDK/Models/BOXRepresentation.m

```diff
@@ -41,27 +46,26 @@ - (instancetype)initWithJSON:(NSDictionary *)JSONResponse
                                                                     nullAllowed:NO];
         self.details = detailsJSON;
         
-        NSDictionary *linksJSON = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyLinks
+        NSDictionary *contentJSON = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyContent
                                                                  inDictionary:JSONResponse
                                                               hasExpectedType:[NSDictionary class]
                                                                   nullAllowed:NO];
         
-        NSDictionary *contentJSON = linksJSON[BOXAPIObjectKeyContent];
-        
-        NSString *contentURLString = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyURL
+        NSString *contentURLString = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyURLTemplate
                                                              inDictionary:contentJSON
                                                           hasExpectedType:[NSString class]
                                                               nullAllowed:NO];
         if (contentURLString.length > 0) {
+            if ([self.type isEqualToString:BOXRepresentationTypeHLS]) {
```

Comment:
I was going to add mp4 checks here, however, down stream we bypass reps and fallback. Will migrate this when we integrate the reset of reps.

As far as setting contentURL to nil, we are ok because URLWithString returns nil for failure. In other words, the result will be nil for all but HLS types.